crates/lsp/src/handlers.rs

```diff
@@ -46,10 +46,13 @@ pub(crate) async fn handle_initialized(
         regs.append(&mut config_diagnostics_regs);
     }
 
-    client
-        .register_capability(regs)
-        .instrument(span.exit())
-        .await?;
+    if !regs.is_empty() {
+        client
+            .register_capability(regs)
+            .instrument(span.exit())
+            .await?;
+    }
+
```

Comment:
Now that we send `initialized()`, this `handle_initialized()` is fired and the server was sending this request to the client that the client test code was not expecting. Since the test client doesn't actually have the capability required here, I guarded it like this to avoid sending the request entirely.

(Note this was quite hard to debug, as it seems like the codec code can't figure out how to deserialize this message from the server at all)